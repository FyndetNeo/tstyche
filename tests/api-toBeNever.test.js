import { expect, test } from "@jest/globals";
import * as tstyche from "tstyche";
import { getFixtureUrl } from "./__utils__/fixtureFactory.js";
import { normalizeOutput } from "./__utils__/normalizeOutput.js";
import { spawnTyche } from "./__utils__/spawnTyche.js";

const fixtureUrl = getFixtureUrl("api-toBeNever");

test("'toBeNever' implementation", () => {
  expect(tstyche.expect).toHaveProperty("type.toBeNever", expect.any(Function));
});

test("toBeNever", async () => {
  const { exitCode, stderr, stdout } = await spawnTyche(fixtureUrl);

  expect(normalizeOutput(stdout)).toMatchSnapshot("stdout");
  expect(stderr).toMatchSnapshot("stderr");

  expect(exitCode).toBe(1);
});
