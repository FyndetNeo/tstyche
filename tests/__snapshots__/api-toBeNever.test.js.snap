// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toBeNever: stderr 1`] = `
"Error: The source type is 'string'.

   7 |   expect(returnsNever()).type.toBeNever();
   8 | 
>  9 |   expect(returnsString()).type.toBeNever();
     |                                ^
  10 | });
  11 | 
  12 | test("is NOT never?", () => {

       at ./__typetests__/toBeNever.tst.ts:9:32 ❭ is never?

Error: The source type is 'never'.

  13 |   expect(returnsString()).type.not.toBeNever();
  14 | 
> 15 |   expect(returnsNever()).type.not.toBeNever();
     |                                   ^
  16 | });
  17 | 

       at ./__typetests__/toBeNever.tst.ts:15:35 ❭ is NOT never?

"
`;

exports[`toBeNever: stdout 1`] = `
"uses TypeScript <<version>> with ./tsconfig.json

fail ./__typetests__/toBeNever.tst.ts
  × is never?
  × is NOT never?

Targets:    1 failed, 1 total
Test files: 1 failed, 1 total
Tests:      2 failed, 2 total
Assertions: 2 failed, 2 passed, 4 total
Duration:   <<timestamp>>

Ran all test files.
"
`;
